scripts/ci/images/ci_prepare_ci_image_on_ci.sh

```diff
@@ -18,4 +18,59 @@
 # shellcheck source=scripts/ci/libraries/_script_init.sh
 . "$( dirname "${BASH_SOURCE[0]}" )/../libraries/_script_init.sh"
 
+# In case GITHUB_REGISTRY_PULL_IMAGE_TAG is different than latest, tries to pull the image indefinitely
+# skips further image checks - assuming that this is the right image
+function wait_for_ci_image_tag {
```

Comment:
This function looks very similar to ``wait_for_prod_images_tag``? Is it on purpose? Is it possible to get rid of this repetitive code?

Reply:
Will take a look.

Reply:
Indeed! It's much simpler now!